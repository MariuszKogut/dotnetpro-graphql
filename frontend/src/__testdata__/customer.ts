export const customerTestData = [
  { id: 1, name: 'Microsoft', location: 'USA' },
  { id: 2, name: 'Oracle', location: 'USA' },
  { id: 3, name: 'IBM', location: 'USA' },
]
