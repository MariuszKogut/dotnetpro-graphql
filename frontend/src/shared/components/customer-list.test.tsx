import React from 'react'
import { render, waitForElementToBeRemoved } from '@testing-library/react'
import CustomerList from './customer-list'
import { customerTestData } from '../../__testdata__/customer'

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
    history: jest.fn(),
  }),
}))

describe('CustomerList', () => {
  it('renders loader', () => {
    const { container } = render(<CustomerList />)
    expect(container).toMatchSnapshot()
  })

  it('renders no data message if no data given', async () => {
    jest.spyOn(window, 'fetch').mockImplementationOnce(() =>
      Promise.resolve({
        text: () => Promise.resolve('[]'),
        status: 200,
      } as Response),
    )

    // Act
    const { container, queryByText } = render(<CustomerList />)

    // Assert
    await waitForElementToBeRemoved(() =>
      queryByText('Daten werden geladen...'),
    )

    expect(container).toMatchSnapshot()
  })

  it('renders data correctly', async () => {
    // Arrange
    jest.spyOn(window, 'fetch').mockImplementationOnce(() =>
      Promise.resolve({
        text: () => Promise.resolve(JSON.stringify(customerTestData)),
        status: 200,
      } as Response),
    )

    // Act
    const { container, queryByText } = render(<CustomerList />)

    // Assert
    await waitForElementToBeRemoved(() =>
      queryByText('Daten werden geladen...'),
    )

    expect(container).toMatchSnapshot()
  })
})
