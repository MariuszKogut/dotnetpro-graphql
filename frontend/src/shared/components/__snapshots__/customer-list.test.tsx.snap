// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerList renders data correctly 1`] = `
<div>
  <div
    class="py-3 col-md-3"
  >
    <div
      class="card card"
    >
      <div
        class="font-weight-bold card-header"
      >
        Microsoft
      </div>
      <div
        class="list-group list-group-flush"
      >
        <div
          class="list-group-item"
        >
          <span
            class="text-muted"
          >
            #
          </span>
           
          1
        </div>
        <div
          class="list-group-item"
        >
          <span
            class="text-muted"
          >
            Location
          </span>
           
          USA
        </div>
      </div>
      <div
        class="d-flex card-body"
      >
        <button
          class="btn btn-secondary"
          type="button"
        >
          Öffnen
        </button>
      </div>
    </div>
  </div>
  <div
    class="py-3 col-md-3"
  >
    <div
      class="card card"
    >
      <div
        class="font-weight-bold card-header"
      >
        Oracle
      </div>
      <div
        class="list-group list-group-flush"
      >
        <div
          class="list-group-item"
        >
          <span
            class="text-muted"
          >
            #
          </span>
           
          2
        </div>
        <div
          class="list-group-item"
        >
          <span
            class="text-muted"
          >
            Location
          </span>
           
          USA
        </div>
      </div>
      <div
        class="d-flex card-body"
      >
        <button
          class="btn btn-secondary"
          type="button"
        >
          Öffnen
        </button>
      </div>
    </div>
  </div>
  <div
    class="py-3 col-md-3"
  >
    <div
      class="card card"
    >
      <div
        class="font-weight-bold card-header"
      >
        IBM
      </div>
      <div
        class="list-group list-group-flush"
      >
        <div
          class="list-group-item"
        >
          <span
            class="text-muted"
          >
            #
          </span>
           
          3
        </div>
        <div
          class="list-group-item"
        >
          <span
            class="text-muted"
          >
            Location
          </span>
           
          USA
        </div>
      </div>
      <div
        class="d-flex card-body"
      >
        <button
          class="btn btn-secondary"
          type="button"
        >
          Öffnen
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CustomerList renders loader 1`] = `
<div>
  <div
    class="col"
  >
    <div
      class="spinner-border"
      role="status"
    >
      <span
        class="sr-only"
      >
        Daten werden geladen...
      </span>
    </div>
  </div>
</div>
`;

exports[`CustomerList renders no data message if no data given 1`] = `
<div>
  <div
    class="col"
  >
    <div
      class="fade alert alert-info show"
      role="alert"
    >
      Keine Daten vorhanden
    </div>
  </div>
</div>
`;
